fsspec/implementations/memory.py

```diff
@@ -164,6 +164,20 @@ def _rm(self, path):
         else:
             raise FileNotFoundError
 
+    def rm(self, path, recursive=False, maxdepth=None):
+        paths = self.expand_path(path, recursive=recursive, maxdepth=maxdepth)
+        for p in reversed(paths):
+            # If the expanded path doesn't exist, it is only because the expanded
+            # path was a directory that does not exist in self.pseudo_dirs. This
+            # is possible if you directly create files without making the
+            # directories first.
+            if not self.exists(p):
```

Comment:
If it cannot happen, we probably shouldn't test for it?

Reply:
Maybe my comment was confusing. I have my comments on the line above, not below. So this if condition can happen, but what cannot happen is checked on line 177 (that `p` is `path`). I can remove that assert if it makes it confusing, I was just adding some defense in case something changes in the future.

Reply:
Please do remove the assert and comment

Reply:
Done!